test/state/host.cpp

```diff
@@ -74,24 +74,38 @@ uint256be Host::get_balance(const address& addr) const noexcept
     return (acc != nullptr) ? intx::be::store<uint256be>(acc->balance) : uint256be{};
 }
 
+namespace
+{
+// For EXTCODE* instructions if the target is an EOF account, then only return EF00.
+// While we only do this if the caller is legacy, it is not a problem doing this
+// unconditionally, because EOF contracts dot no have EXTCODE* instructions.
+bytes_view extcode(bytes_view code) noexcept
+{
+    if (code.size() >= 2 && code[0] == 0xEF && code[1] == 0x00)
+        return code.substr(2);
```

Comment:
Ah this is actually broken, need to get first 2 bytes, not exclude first 2 bytes.

Reply:
```suggestion
        return code.substr(0, 2);
```